memory_set/src/set.rs

```diff
@@ -1,16 +1,18 @@
-use alloc::{collections::BTreeMap, vec::Vec};
+use alloc::collections::BTreeMap;
+#[allow(unused_imports)] // this is a weird false alarm
+use alloc::vec::Vec;
 use core::fmt;
 
-use memory_addr::{VirtAddr, VirtAddrRange};
+use memory_addr::AddrRange;
 
 use crate::{MappingBackend, MappingError, MappingResult, MemoryArea};
 
 /// A container that maintains memory mappings ([`MemoryArea`]).
-pub struct MemorySet<F: Copy, P, B: MappingBackend<F, P>> {
-    areas: BTreeMap<VirtAddr, MemoryArea<F, P, B>>,
+pub struct MemorySet<B: MappingBackend> {
+    areas: BTreeMap<usize, MemoryArea<B>>,
```

Comment:
When `MemoryAddr` impl `Ord`, can `B::Addr` be used as a key here?

Reply:
Yes